raiden/network/proxies/service_registry.py

```diff
@@ -75,6 +86,27 @@ def get_service_url(
             return None
         return result
 
+    def current_price(self, block_identifier: BlockSpecification) -> int:
+        """Gets the currently required deposit amount."""
+
+        return self.proxy.contract.functions.currentPrice().call(block_identifier=block_identifier)
+
+    def token_address(self, block_identifier: BlockSpecification) -> AddressHex:
+        return self.proxy.contract.functions.token().call(block_identifier=block_identifier)
+
+    def deposit(self, limit_amount: int) -> None:
```

Comment:
Do we need this here? Only the services should ever care, right?

Reply:
There is a test case in the client that tests `setURL()`.  Now a deposit is needed before `setURL()`.